### ISSUE TITLE
Batman: Arkham Origins - no smoke

### ISSUE THREAD

**Issue by cakesale** [daniel-schuermann/mesa#142]
No smoke is present after throwing a smoke pellet when aco is active. Smoke is also missing in the main menu screens.

I have tested with RADV_PERFTEST=llvm and confirmed there is smoke.

I have also tested with various kernels and mesa drivers confirming aco is the issue.

Otherwise the game runs GREAT from benchmarks and small amount of play!

**Comment by pendingchaos** [comment]
What GPU and ACO version are you using?

Are you able to get a Renderdoc capture of the issue (it's probably better to capture the main menu than in-game)?

**Comment by cakesale** [comment]
Sorry I'm a bit sick at the moment so not thinking straight.

Arch Linux
CPU: Intel Core i7-6700K @ 8x 4.2GHz
GPU: Radeon RX 580 Series (POLARIS10, DRM 3.33.0, 5.3.1-6.1-tkg-pds, LLVM 8.0.1)

Repository      : valveaur
Name            : mesa-aco-git
Version         : 19.3.0_devel.20190919.b738bf06917-3

Saw something about Renderdoc after I had posted the issue. I'll try and do some research tomorrow and try and see if I can submit something. Sorry for useless information in last reply lol.

**Comment by daniel-schuermann** [comment]
No worries :)

You can get instructions on how to create a renderdoc capture if you click `New Issue` and `Get Started` on a Bug report. (Please don't submit a new issue, though :P )

**Comment by cakesale** [comment]
Ok, so I've tried creating a capture but for the life of me can't get it to work.

The only game that seems to want to launch and be able to capture is Dota 2.

For Batman the game crashes if trying to use "ENABLE_VULKAN_RENDERDOC_CAPTURE=1 %command%", "RADV_PERFTEST=llvm ENABLE_VULKAN_RENDERDOC_CAPTURE=1 %command%" as well as running Steam with "env ENABLE_VULKAN_RENDERDOC_CAPTURE=1 steam".

I can launch other native and Windows games, some with different Proton versions but none seem to allow me to do a trace other than Dota 2...

edit: just did a test with Rise of the Tomb Raider and it is able to perform a capture.

**Comment by SR-dude** [comment]
I own the game. I never played it however. LLVM did not offer any noticeable differences.
If saves are easily transferable, can you give me one?  If not then a screen shot of the main menu
would do, with a note from you what I should look for.  I can't find anything wrong with the main menu.

**Comment by cakesale** [comment]
I went to test last night before going out so I could take some screenshots and I wasn't getting the issue anymore... I had recently installed a new mesa with aco branch from another pkgbuild and it wasn't being affected like before.

This morning there was a new mesa-aco from valveaur repo and can confirm no longer an issue with this new build (19.3.0_devel.20190927.d0ed7ce952c-4).

I managed to take some screenshots before I upgraded to latest build on valveaur repo.

![menu 1](https://user-images.githubusercontent.com/5090375/65814800-0e974380-e1e7-11e9-853f-88b6063951e8.jpg)
![menu 1 aco](https://user-images.githubusercontent.com/5090375/65814801-0e974380-e1e7-11e9-8cff-6898c9b0c0de.jpg)
![smoke](https://user-images.githubusercontent.com/5090375/65814802-0e974380-e1e7-11e9-994f-73e78d04f130.jpg)
![smoke none - aco](https://user-images.githubusercontent.com/5090375/65814803-0f2fda00-e1e7-11e9-8066-a4f0ef038832.jpg)
